commands/account/auth.ts

```diff
@@ -0,0 +1,212 @@
+  deleteEmptyConfigFile,
+  getConfigDefaultAccount,
+} from '@hubspot/local-dev-lib/config';
+import { getAccountIdentifier } from '@hubspot/local-dev-lib/config/getAccountIdentifier';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import {
+  getAccessToken,
+  updateConfigWithAccessToken,
+} from '@hubspot/local-dev-lib/personalAccessKey';
+import { ENVIRONMENTS } from '@hubspot/local-dev-lib/constants/environments';
+import { toKebabCase } from '@hubspot/local-dev-lib/text';
+import { Environment } from '@hubspot/local-dev-lib/types/Config';
+import { CLIAccount } from '@hubspot/local-dev-lib/types/Accounts';
+import { PERSONAL_ACCESS_KEY_AUTH_METHOD } from '@hubspot/local-dev-lib/constants/auth';
+import { DEFAULT_HUBSPOT_CONFIG_YAML_FILE_NAME } from '@hubspot/local-dev-lib/constants/config';
+
+import { addGlobalOptions, addTestingOptions } from '../../lib/commonOpts';
+import { handleExit } from '../../lib/process';
+import { debugError } from '../../lib/errorHandlers/index';
+import { i18n } from '../../lib/lang';
+import { trackCommandUsage, trackAuthAction } from '../../lib/usageTracking';
+import { personalAccessKeyPrompt } from '../../lib/prompts/personalAccessKeyPrompt';
+import { cliAccountNamePrompt } from '../../lib/prompts/accountNamePrompt';
+import { setAsDefaultAccountPrompt } from '../../lib/prompts/setAsDefaultAccountPrompt';
+import { EXIT_CODES } from '../../lib/enums/exitCodes';
+import { uiFeatureHighlight } from '../../lib/ui';
+import { CommonArgs, ConfigArgs } from '../../types/Yargs';
+
+const i18nKey = 'commands.account.subcommands.auth';
+
+const TRACKING_STATUS = {
+  STARTED: 'started',
+  ERROR: 'error',
+  COMPLETE: 'complete',
+};
+
+async function createOrUpdateConfig(
+  env: Environment,
+  doesConfigExist: boolean,
+  disableTracking: boolean | undefined,
+  authType: string,
+  account?: number
+): Promise<CLIAccount | null> {
+  try {
+    const { personalAccessKey } = await personalAccessKeyPrompt({
+      env,
+      account,
+    });
+    const token = await getAccessToken(personalAccessKey, env);
+    const defaultName = token.hubName ? toKebabCase(token.hubName) : undefined;
+
+    const name = doesConfigExist
+      ? undefined
+      : (await cliAccountNamePrompt(defaultName)).name;
+
+    const updatedConfig = await updateConfigWithAccessToken(
+      token,
+      personalAccessKey,
+      env,
+      name,
+      !doesConfigExist
+    );
+
+    if (!updatedConfig) return null;
+
+    if (doesConfigExist && !updatedConfig.name) {
+      updatedConfig.name = (await cliAccountNamePrompt(defaultName)).name;
+      updateAccountConfig({
+        ...updatedConfig,
+      });
+      writeConfig();
+    }
+
+    return updatedConfig;
+  } catch (e) {
+    if (!disableTracking) {
+      await trackAuthAction('account-auth', authType, TRACKING_STATUS.ERROR);
+    }
+    debugError(e);
+    return null;
+  }
+}
+
+export const describe = ''; // i18n(`${i18nKey}.describe`);
```

Comment:
This needs to be `undefined` otherwise this command will show up when a user runs `hs account --help`